build/gulpfile.vscode.darwin.js

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+const gulp = require('gulp');
+const { signAsync } = require('electron-osx-sign');
+const util = require('./lib/util');
+const path = require('path');
+
+gulp.task('vscode-codesign', async () => {
+	const buildDir = process.env['AGENT_BUILDDIRECTORY'];
+
+	if (!buildDir) {
+		throw new Error('$AGENT_BUILDDIRECTORY not set');
+	}
+
+	if (process.env.VSCODE_QUALITY !== 'insider' && process.env.VSCODE_QUALITY !== 'stable') {
+		return;
+	}
+
+	const appRoot = path.join(buildDir, 'VSCode-darwin');
+	const appName = process.env.VSCODE_QUALITY === 'insider' ? 'Visual Studio Code - Insiders.app' : 'Visual Studio Code.app';
+	const appFrameworkPath = path.join(appRoot, appName, 'Contents', 'Frameworks');
+	const helperAppBaseName = process.env.VSCODE_QUALITY === 'insider' ? 'Code - Insiders' : 'Code';
+	const gpuHelperAppName = helperAppBaseName + ' Helper (GPU).app';
+	const pluginHelperAppName = helperAppBaseName + ' Helper (Plugin).app';
+	const rendererHelperAppName = helperAppBaseName + ' Helper (Renderer).app';
+
+	const defaultOpts = {
+		app: path.join(appRoot, appName),
+		platform: 'darwin',
+		entitlements: path.join(__dirname, 'azure-pipelines', 'darwin', 'app-entitlements.plist'),
+		hardenedRuntime: true,
+		'pre-auto-entitlements': false,
+		'pre-embed-provisioning-profile': false,
+		keychain: path.join(buildDir, 'buildagent.keychain'),
+		version: util.getElectronVersion(),
+		identity: '99FM488X57',
```

Comment:
Yeah I believe this is correct, we'd have to see how this translates to the actual `codesign` cli command, to compare to what was happening before.

Reply:
Is maybe the new thing calling codesign under a different user?

Reply:
There seems to be a [`debuglog` utility](https://github.com/electron/electron-osx-sign/blob/63d7b60cfae6e2c830f7e2bda8d376d547544c85/sign.js#L385), let's enable it and see if we learn more: https://github.com/electron/electron-osx-sign#debug

Reply:
https://monacotools.visualstudio.com/Monaco/_build/results?buildId=75004